content/articles/abstractive-summarization-using-deep-learning/index.md

```diff
@@ -0,0 +1,175 @@
+
+All that's left to do is to perform abstractive summarization on some Wikipedia text.
+
+### Performing abstractive summarization 
+This last step involves taking a bunch of text, passing it through the Pegasus model, and seeing how the model performs abstractive summarization on the text.
+
+Let's find some text that we want to summarize. Let's create a variable called `text` and add some text to it.
+
+```python
+text = """
+Hugging Face is a great open-source library doing powerful work in the Natural Language Processing (NLP) space. The library has a bunch of pre-trained models which you can leverage or fine-tune. 
+
+The library has many models, including BERT and GPT-2 models that perform various tasks, but we’ll be leveraging the pre-trained language pipeline for our purpose. Rather than going ahead and training a huge language model such as GPT-2 with 1.5 billion parameters, one can leverage the ML pipeline instead."""
+```
+
+We've picked some text from this [tutorial](/engineering-education/building-a-simple-translation-app-using-python-for-beginners/) on Section's website.
+
+Using our tokenizer, which we loaded earlier, let's convert our text into its token representation.
+
+```python
+tokens = tokenizer_model(text, truncation=True, padding="longest", return_tensors="pt")
+```
+We've created a variable called `tokens` to store our token representation. We use our tokenizer `tokenizer_model`, which we created earlier, to tokenize our texts. The `truncation=True` argument allows the model to truncate our texts into a size suitable for input into our model. The `return_tensors` argument tells the model to use tensors from PyTorch.
+
+Let's view our tokens. We achieve this by writing the following command:
+
+```python
+tokens
+```
+Let's go ahead and try to summarize this text.
+
+```python
+summary = loaded_model.generate(**tokens)
+```
+The `**tokens` unpacks our tokens and pass them into our model. The asterisks in `**tokens` are simply adding the `input_ids` and `attention_mask` present in the results above. 
+
+Let's check our generated summary results. We do this by typing the following command:
+
+```python
+summary
+```
+These results represent our output tensors. This is our summary in tokens.
+
+These may be just a bunch of numbers to humans, but not to machines. This is how they understand language. Decoding these values will help us make sense of these numbers.
+
+```python
+tokenizer_model.decode(summary[0])
+```
+The `summary` result above shows that the results are in a nested list. But we only need the first result, so that's why we've indexed [0].
+
+Running that command shows our summarized text. 
+
+Impressive, right?
+
+All that big block of text summarized into that short text. 
+
+So, here's the actual test. If you go through the huge block of text, you'll find that this summarized version of the text doesn't exist. It's novel and has been completely generated by the model.
```

Comment:
The beginning of this sentence is a bit unclear. A reader will likely ask, what exactly is being tested?